test/html/pipeline/mention_filter_test.rb

```diff
@@ -72,6 +72,34 @@ def test_links_to_more_info_when_info_url_given
       filter(body, '/', 'https://github.com/blog/821').to_html
   end
 
+  def test_base_url_slash
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '/').to_html
+  end
+
+  def test_base_url_under_custom_route
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/userprofile/jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '/userprofile').to_html
+  end
+
+  def test_base_url_tilde
+    body = "<p>Hi, @jch!</p>"
+    link = "<a href=\"/~jch\" class=\"user-mention\">@jch</a>"
+    assert_equal "<p>Hi, #{link}!</p>",
+      filter(body, '~').to_html
+  end
```

Comment:
You're welcome.  :smile: 

In this case, the input is from a gem user, and it's a configuration value.  If a regular user sent the input, it might be a different story.  

Thank you for applying the commit and the Pull Request!  :clap: